### ISSUE TITLE
In Page Navigation (relative links on the same page)

### ISSUE THREAD

**Issue by kunalshetye** [sdl/dxa-web-application-java#53]
When generating Component Links, if the component link resolves to the page itself then can it be generated as a samepage.html#1234 where 1234 is the Item Id of Component? For this to work the LinkResolver needs to be aware of the PageContext.

We have a use case where on a page we have a Link List with Component Links and the Component Links are pointing to the Component Presentations on the same page.

What the we are trying to achieve is generate "relative" links on the page that allow the users to navigate across different sections (represented by different component presentations) on the same page.

Does the use case make sense?

**Comment by rpannekoek** [comment]
Yes, the use case makes sense.  Labeled this as a future Enhancement.

**Comment by rpannekoek** [comment]
TSI-1939 (internal issue ID for tracking purposes)

**Comment by rpannekoek** [comment]
Internal Issue ID (for tracking purposes): CRQ-12780
